src/rgw/rgw_pubsub.h

```diff
@@ -629,7 +635,11 @@ class RGWPubSub
   // get a topic with by its name and populate it into "result"
   // return -ENOENT if the topic does not exists 
   // return 0 on success, error code otherwise
-  int get_topic(const DoutPrefixProvider *dpp, const std::string& name, rgw_pubsub_topic& result, optional_yield y) const;
+  int get_topic(const DoutPrefixProvider* dpp,
+                const std::string& name,
+                rgw_pubsub_topic& result,
+                optional_yield y,
+                bool fetch_bucket_mapping = false) const;
```

Comment:
probably better not to use a default here. so we are explicit on where we need the mapping and where we don't need it

Reply:
> probably better not to use a default here. so we are explicit on where we need the mapping and where we don't need it

Done.